lib/travis/logs/app.rb

```diff
@@ -126,6 +127,33 @@ def initialize(auth_token: ENV['AUTH_TOKEN'].to_s,
         body nil
       end
 
+      get '/log-parts/:job_id' do
+        halt 500, 'authentication token is not set' if auth_token.empty?
+        halt 403 unless authorized?(request)
+
+        job_id = Integer(params[:job_id])
+        part_numbers = if params[:part_numbers].nil?
+                         []
+                       else
+                         params[:part_numbers].split(',').map(&:to_i)
+                       end
+        after = if params[:after].nil?
```

Comment:
similar suggestion here:
``` ruby
after = params[:after]
after &&= Integer(after) # this keeps it as nil if it's nil, otherwise it casts it as an Integer
```